### PR TITLE
added stats for data events

### LINKED ISSUE #1003: Can't see data events stats/metrics
When running tetragon with metrics ( --metrics-server=:2112)
I can't see any stats about data events, which we clearly posted.

We might be missing support to peg data events (ops.MSG_OP_DATA).

### PR DESCRIPTION
**Pull request by swastik959**
Fixes #1003

### CODE REVIEW

File: pkg/observer/data_stats.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+const (
+	EventReceived EventType = iota
+	EventMatched
+	EventNotMatched
+)
+
+var EventTypeStrings = map[EventType]string{
+	EventReceived:   "EventReceived",
+	EventMatched:    "EventMatched",
+	EventNotMatched: "EventNotMatched",
```
Comment (olsajiri) [inline pkg/observer/data_stats.go:29]: the final output already says it's for data event, so let's keep this just with the action name, like:

jolsa@ubuntu-22:~/tetragon$ curl http://localhost:8080/metrics 2>&1 | grep ^tetragon_data_event_stats
tetragon_data_event_stats{event="Added"} 14
tetragon_data_event_stats{event="Matched"} 14
tetragon_data_event_stats{event="Received"} 14

s/EventReceived/Received/ ...

**Review by olsajiri — CHANGES_REQUESTED** [review]
I pushed suggested comments into this commit, please check:
https://github.com/cilium/tetragon/commit/2d7c1f8e75fad8a3bc7feebbb0c6e1fa1b242b2b

File: pkg/observer/data_stats.go (hunk 1 above)
Comment (swastik959) [reply pkg/observer/data_stats.go:29]: @olsajiri hi sorry for the delay but I am unable to understand what you want to say

**Comment by swastik959** [comment]
@olsajiri I have made some changes please review them

File: pkg/observer/data_stats.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import (
+	"github.com/cilium/tetragon/pkg/metrics/consts"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	// Define a counter metric for data event statistics
+	DataEventStats = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name:        consts.MetricNamePrefix + "data_event_stats",
```
Comment (lambdanis) [inline pkg/observer/data_stats.go:12]: Can you add a `_total` suffix to the counter, for compatibility with the [OpenMetrics](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#suffixes) conventions?

**Comment by mtardy** [comment]
Hey @swastik959, do you still plan to work on this or should we close this PR for now? Thanks!!

**Comment by swastik959** [comment]
@mtardy sure will complete it

**Comment by lambdanis** [comment]
`tetragon_data_events_total` and `tetragon_data_event_size` metrics were eventually added in #1224 and refactored later. They seem to behave, so I'm closing this PR. Thanks for opening it @swastik959.

**Pull request closed without merging**